src/Symfony/Component/DependencyInjection/Compiler/PassConfig.php

```diff
@@ -71,6 +71,12 @@ public function __construct()
             new CheckArgumentsValidityPass(false),
         ));
 
+        $this->beforeRemovingPasses = array(
+            -100 => array(
+                new ResolvePrivatesPass(),
```

Comment:
This could also be the first removing pass instead.

Reply:
which one is better? semantically, this is not a removing pass.
Is there any technical reason that would help decide?
otherwise, I think we should leave as is, isn't it?

Reply:
Well, "optimization" passes are not really about optimizing either 😄